### DISCUSSION TITLE
How to Define and Handle Custom Methods in WalletConnect 2.0 Using Web3Wallet? (eip155)

### DISCUSSION THREAD

**Discussion by ramvi** [WalletConnect/walletconnect-monorepo#2805] [Developer Support] [answered]
Hello, I am currently migrating from `sign-client` to `Web3Wallet` in WalletConnect 2.0. I've understood the main steps for the migration, but I am unsure about how to handle custom methods. 

In the older version, I used to define custom methods like this:
```javascript
const { uri, approval } = await client.connect({
    requiredNamespaces: {
        eip155: {
            methods: ["request_credential", "present_credential", "receive_credential"],
            chains: ["eip155:5"],
            events: [],
        },
    },
});
```
Could you please provide some guidance or examples on how to define and handle custom methods similar to this using the `Web3Wallet` library? Thank you in advance for your help.

**Comment by finessevanes** [comment, marked as answer]
@ramvi this should help you get going: https://docs.walletconnect.com/2.0/web/web3wallet/wallet-usage#session-approval

You'll want to be sure to use [Namespaces builder util](https://docs.walletconnect.com/2.0/web/web3wallet/wallet-usage#-namespaces-builder-util)

**Reply by ramvi** [reply]
Thank you 🙏
